### PR TITLE
Bugfix: Fix delivery action in tray loader

### LINKED ISSUE #351: Delivery not working in 0.3.0-dev.1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior:

I have a delivery setup like this:

```
{
  "name": "main",
  "directory": "{root[work]}/{project[name]}/delivery/{yyyy}-{mm}-{dd}/{hierarchy}/{folder[name]}/",
  "file": "{project[code]}_{folder[name]}_{task[short]}_{@version}<_{output}><.{@frame}>.{ext}"
}
```
but when I try to delivery I'm getting this error:
```
  File "C:\Users\jacob\Syncthing\Projekt\Ayon\ayon-core\client\ayon_core\plugins\load\delivery.py", line 285, in _get_templates
    path_template = value["path"]
KeyError: 'path'

```

It looks like `path` should be `directory` from what I can see in the code https://github.com/ynput/ayon-core/blob/85e58545af4b4784b0f73e9126434bfb0fe30263/client/ayon_core/plugins/load/delivery.py#L288C24-L288C37
If I change it `path` to `directory` I'm now getting this error:

```
Traceback (most recent call last):
  File "C:\Users\jacob\Syncthing\Projekt\Ayon\ayon-core\client\ayon_core\plugins\load\delivery.py", line 49, in load
    dialog = DeliveryOptionsDialog(contexts, self.log)
  File "C:\Users\jacob\Syncthing\Projekt\Ayon\ayon-core\client\ayon_core\plugins\load\delivery.py", line 166, in __init__
    self._update_template_value()
  File "C:\Users\jacob\Syncthing\Projekt\Ayon\ayon-core\client\ayon_core\plugins\load\delivery.py", line 345, in _update_template_value
    self.template_label.setText(template_value)
TypeError: 'PySide2.QtWidgets.QLabel.setText' called with wrong argument types:
  PySide2.QtWidgets.QLabel.setText(dict)
Supported signatures:
  PySide2.QtWidgets.QLabel.setText(str)
```

So if I change `self.template_label.setText(template_value)` to `self.template_label.setText(template_value["directory"])` in https://github.com/ynput/ayon-core/blob/85e58545af4b4784b0f73e9126434bfb0fe30263/client/ayon_core/plugins/load/delivery.py#L353C32-L353C49 the delivery ui works as expected

### Expected Behavior:

_No response_

### Version

1.0.0

### What platform you are running on?

Windows

### Steps To Reproduce:

Add a delivery preset.
Try to deliver something in the Tray Loader

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the bug report.

### Relevant log output:

_No response_

### Additional context:

_No response_

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Fix delivery action after recent refactoring of ayon core.

## Additional info

Fixes #351 

Since directory and file is now separate in the delivery templates I've also made them two labels in the delivery UI for better clarity as to what is what.

image

## Testing notes:

1. Set up a delivery template in Anatomy settings `ayon+anatomy://PROJECTNAME/templates/delivery`, for example like these examples:

```json
[
  {
    "name": "client",
    "directory": "{root[work]}/{folder[name]}/{product[name]}",
    "file": "{folder[name]}_{product[name]}_{@version}<_{output}><.{@frame}>.{ext}"
  },
  {
    "name": "previews",
    "directory": "{root[work]}",
    "file": "{folder[name]}_{product[name]}_{@version}<_{output}><.{@frame}>.{ext}"
  },
  {
    "name": "per_asset",
    "directory": "{root[work]}/{folder[name]}",
    "file": "{folder[name]}_{product[name]}_{@version}<_{output}><.{@frame}>.{ext}"
  },
  {
    "name": "main",
    "directory": "{root[work]}/{project[name]}/delivery/{yyyy}-{mm}-{dd}/{hierarchy}/{folder[name]}/",
    "file": "{project[code]}_{folder[name]}_{task[short]}_{@version}<_{output}><.{@frame}>.{ext}"
  }
]
```

2. In Tray Launcher open the Loader
3. Using the `Delivery` action in the tray loader initiate a delivery to an output root.

### CODE REVIEW

File: client/ayon_core/plugins/load/delivery.py [hunk 1]
```diff
@@ -123,7 +129,8 @@ def __init__(self, contexts, log=None, parent=None):
 
         input_layout.addRow("Selected representations", selected_label)
         input_layout.addRow("Delivery template", dropdown)
-        input_layout.addRow("Template value", template_label)
+        input_layout.addRow("Template directory", template_dir_label)
+        input_layout.addRow("Template file", template_file_label)
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/load/delivery.py:133]:
```suggestion
        input_layout.addRow("Directory template", template_dir_label)
        input_layout.addRow("File template", template_file_label)
```

File: client/ayon_core/plugins/load/delivery.py [hunk 2]
```diff
@@ -282,10 +290,10 @@ def _get_templates(self, anatomy):
         """Adds list of delivery templates from Anatomy to dropdown."""
         templates = {}
         for template_name, value in anatomy.templates["delivery"].items():
-            path_template = value["path"]
+            directory_template = value["directory"]
             if (
-                not isinstance(path_template, str)
-                or not path_template.startswith('{root')
+                not isinstance(directory_template, str)
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/load/delivery.py:295]: I guess the `str` type check is not necessary anymore.

Comment (BigRoy) [reply client/ayon_core/plugins/load/delivery.py]: Should be resolved.

**Comment by LiborBatek** [comment]
Meanwhile been successfully tested :)

Delivery templates created:

Screenshot 2024-04-09 153131

and in the Loader 

Screenshot 2024-04-09 153617

Been tested with those 3 templates, all working well!

**Pull request merged**